backend/src/Services/Account/AccountService.ts

```diff
@@ -82,6 +82,13 @@ export default class AccountService {
             const key = customKey ? customKey : this.publicKey;
             return AccountServiceUtils.extractUserFromRequest(req, token, key);
         }
+        const tempId = req.cookies.temp_id;
+        if(tempId){
+            return {
+                'typ': 'TEMP_ID', // keycloak populates typ with ID, so I'm trying to follow a similar convention
+                'sub': tempId
+            }
+        }
         return null
```

Comment:
One thing to look out for here is places where we are expecting to return null in the user is not logged in, like the expectUserFromRequest function in controllerUtils.ts. 

You might also want to add the temp_id cookie to the backend test requests as a default to see if that breaks anything.

Reply:
Good point

controllerUtils.ts doesn't seem to be used so I removed it to avoid confusion. I didn't see any similar assumptions with other calls to extractUserFromRequest

I also added a bunch of tests checking for temp user scenarios